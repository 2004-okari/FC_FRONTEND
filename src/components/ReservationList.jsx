import { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { displayReservation } from '../redux/reservation/reservationSlice';
import { getBikes } from '../redux/Bikes/bikeSlice';

const ReservationList = () => {
  const dispatch = useDispatch();
  const { display, isLoading } = useSelector((store) => store.reservation);
  const { message } = useSelector((store) => store.bikes);

  useEffect(() => {
    dispatch(getBikes());
    dispatch(displayReservation());
  }, [dispatch]);

  if (isLoading) {
    return (
      <div className="flex items-center justify-center w-full">Loading</div>
    );
  }

  if (display) {
    return (
      <div className="flex flex-col gap-6 p-4 items-center w-full">
        <h1 className="font-bold text-2xl mb-8">My Reservatons</h1>
        {display.message.map((reserve) => {
          const bike = message.bikes && message.bikes.find((bike) => bike.id === reserve.bike_id);
          return (
            <div className="flex gap-3 border w-full  md:w-3/4 p-4 items-center justify-between shadow-xl rounded-xl" key={reserve.id}>
              <h2 className="font-bold">{bike && bike.name}</h2>
              <p>{reserve.city}</p>
              <p>{reserve.date}</p>
            </div>
          );
        })}
      </div>
    );
  }
  return (
    <div>No Reservation to show</div>
  );
};

export default ReservationList;
