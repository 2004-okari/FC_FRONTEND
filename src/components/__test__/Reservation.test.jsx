import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import ReservationList from '../ReservationList';

const mockStore = configureStore([]);

describe('ReservationList Component', () => {
  let store;
  let component;

  beforeEach(() => {
    store = mockStore({
      reservation: {
        display: {
          message: [
            {
              id: 1, bike_id: 1, city: 'City1', date: '2023-12-31',
            },
            {
              id: 2, bike_id: 2, city: 'City2', date: '2023-12-30',
            },
          ],
        },
        isLoading: false,
      },
      bikes: {
        message: {
          bikes: [
            { id: 1, name: 'Bike1' },
            { id: 2, name: 'Bike2' },
          ],
        },
      },
    });

    component = renderer.create(
      <Provider store={store}>
        <ReservationList />
      </Provider>,
    );
  });

  it('renders correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
