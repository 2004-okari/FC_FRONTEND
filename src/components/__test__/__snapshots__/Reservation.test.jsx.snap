// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReservationList Component renders correctly 1`] = `
<div
  className="flex flex-col gap-6 p-4 items-center w-full"
>
  <h1
    className="font-bold text-2xl mb-8"
  >
    My Reservatons
  </h1>
  <div
    className="flex gap-3 border w-full  md:w-3/4 p-4 items-center justify-between shadow-xl rounded-xl"
  >
    <h2
      className="font-bold"
    >
      Bike1
    </h2>
    <p>
      City1
    </p>
    <p>
      2023-12-31
    </p>
  </div>
  <div
    className="flex gap-3 border w-full  md:w-3/4 p-4 items-center justify-between shadow-xl rounded-xl"
  >
    <h2
      className="font-bold"
    >
      Bike2
    </h2>
    <p>
      City2
    </p>
    <p>
      2023-12-30
    </p>
  </div>
</div>
`;
